### ISSUE TITLE
Non-portable LDFLAGS in libpolys/polys/Makefile.am

### ISSUE THREAD

**Issue by orlitzky** [Singular/Singular#1127]
At https://github.com/Singular/Singular/blob/spielwiese/libpolys/polys/Makefile.am#L72 we have,

```m4
P_PROCS_MODULE_LDFLAGS = -shared -module -dynamic -export-dynamic -avoid-version -weak_reference_mismatches weak -undefined dynamic_lookup -flat_namespace
```

and several of those are non-portable. For example `-weak_reference_mismatch` and `-flat_namespace` are Darwin linker options. (I'm not sure what `weak` does?) This caused a build failure reported at https://bugs.gentoo.org/778563

In https://github.com/Singular/Singular/blob/spielwiese/m4/p-procs.m4, the `AX_APPEND_LINK_FLAGS` macro is used to append only those flags that are supported by the linker. Maybe the same can be done for the `P_PROCS_MODULE_LDFLAGS` variable?

**Comment by hannes14** [comment]
fixed with 760a3448dba802a0197f4944a864d32632e30aff

**Comment by orlitzky** [comment]
Thank you!

**Comment by orlitzky** [comment]
Now that I've tried this in real life, it looks like these flags are getting added back in by any dyn_modules the user chooses to build, e.g. https://github.com/Singular/Singular/blob/spielwiese/Singular/dyn_modules/Order/Makefile.am#L22

Maybe it's as simple as duplicating the  `AX_APPEND_LINK_FLAGS(..., [P_PROCS_MODULE_LDFLAGS])` to the top-level build system?

**Comment by orlitzky** [comment]
Was https://github.com/Singular/Singular/commit/1db01d01563216ec9e9300e1eaa67ed79cc937f6 meant to address this? It does partially, by eliminating the failure due to `-weak_reference_mismatches`, but the `-flat_namespace` one still unfortunately remains. For example on Gentoo with slibtool (which won't ignore the error):

```
rlibtool: link: x86_64-pc-linux-gnu-g++ .libs/subsets_la-subsets.o -O2 -pipe -march=native -fno-common -flat_namespace -Wl,-O1 -Wl,--as-needed -pipe -fno-common -rdynamic -Wl,-undefined,dynamic_lookup -lcddgmp -lreadline -lmpfr -lgmp -lrt -shared -fPIC -Wl,-soname -Wl,subsets.so -o .libs/subsets.so -Wl,--export-dynamic
x86_64-pc-linux-gnu-g++: error: unrecognized command-line option '-flat_namespace'
```

I wish I could be of more use debugging this (since it's a niche problem) but I have no access to macOS.

**Comment by hannes14** [comment]
should be fixed with https://github.com/Singular/Singular/commit/8300693478181d3f9e7be72a7cd1bd5dd3024c8a

**Comment by orlitzky** [comment]
> should be fixed with [8300693](https://github.com/Singular/Singular/commit/8300693478181d3f9e7be72a7cd1bd5dd3024c8a)

Indeed, thank you!
